internal/kafka/internal/handlers/data_plane_cluster_test.go

```diff
@@ -0,0 +1,285 @@
+	}
+
+	RegisterTestingT(t)
+
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			h := NewDataPlaneClusterHandler(tt.fields.dataplaneClusterService)
+			err := h.validateBody(tt.args.request)()
+			Expect(err != nil).To(Equal(tt.wantErr))
+		})
+	}
+}
+
+func Test_validateStrimzi(t *testing.T) {
+	type fields struct {
+		dataplaneClusterService services.DataPlaneClusterService
+	}
+
+	type args struct {
+		request *private.DataPlaneClusterUpdateStatusRequest
+	}
+
+	tests := []struct {
+		name    string
+		args    args
+		fields  fields
+		wantErr bool
+	}{
+		{
+			name: "should return no error with valid Strimzi in the dataplane cluster status",
+			args: args{
+				request: dataplanemocks.BuildValidDataPlaneClusterUpdateStatusRequest(nil),
+			},
+			fields: fields{
+				dataplaneClusterService: &services.DataPlaneClusterServiceMock{},
+			},
+			wantErr: false,
+		},
+		{
+			name: "should return an error with empty Strimzi Version",
+			args: args{
+				request: dataplanemocks.BuildValidDataPlaneClusterUpdateStatusRequest(func(statusRequest *private.DataPlaneClusterUpdateStatusRequest) {
+					statusRequest.Strimzi = []private.DataPlaneClusterUpdateStatusRequestStrimzi{
+						{
+							Ready:            true,
+							Version:          "",
+							KafkaVersions:    []string{"2.8.0"},
+							KafkaIbpVersions: []string{"2.8.0"},
+						},
+					}
+				}),
+			},
+			fields: fields{
+				dataplaneClusterService: &services.DataPlaneClusterServiceMock{},
+			},
+			wantErr: true,
+		},
+		{
+			name: "should return an error with empty KafkaVersions",
+			args: args{
+				request: dataplanemocks.BuildValidDataPlaneClusterUpdateStatusRequest(func(statusRequest *private.DataPlaneClusterUpdateStatusRequest) {
+					statusRequest.Strimzi = []private.DataPlaneClusterUpdateStatusRequestStrimzi{
+						{
+							Ready:            true,
+							Version:          "2.8.0",
+							KafkaVersions:    []string{},
+							KafkaIbpVersions: []string{"2.8.0"},
+						},
+					}
+				}),
+			},
+			fields: fields{
+				dataplaneClusterService: &services.DataPlaneClusterServiceMock{},
+			},
+			wantErr: true,
+		},
+		{
+			name: "should return an error with empty KafkaIbpVersions",
+			args: args{
+				request: dataplanemocks.BuildValidDataPlaneClusterUpdateStatusRequest(func(statusRequest *private.DataPlaneClusterUpdateStatusRequest) {
+					statusRequest.Strimzi = []private.DataPlaneClusterUpdateStatusRequestStrimzi{
+						{
+							Ready:            true,
+							Version:          "2.8.0",
+							KafkaVersions:    []string{"2.8.0"},
+							KafkaIbpVersions: []string{},
+						},
+					}
+				}),
+			},
+			fields: fields{
+				dataplaneClusterService: &services.DataPlaneClusterServiceMock{},
+			},
+			wantErr: true,
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			h := NewDataPlaneClusterHandler(tt.fields.dataplaneClusterService)
```

Comment:
Let's add a `t.Parallel()` call here for parallel test execution

Reply:
done